mlflow/metrics/metric_definitions.py

```diff
@@ -33,6 +33,28 @@ def _validate_text_data(data, metric_name, column_name):
     return True
 
 
+def _validate_and_fix_text_tuple_data(data, metric_name, column_name):
+    """Validates that the data is a list of a tuple of strings and is non-empty"""
+    if data is None or len(data) == 0:
+        return False
+
+    for row, tup in enumerate(data):
+        if not isinstance(tup, tuple) or not all(isinstance(val, str) for val in tup):
+            if isinstance(tup, str):
+                # Single entry tuples get unpacked.
+                # So if the entry is a string, put them back into a tuple.
+                data[row] = (tup,)
```

Comment:
https://github.com/mlflow/mlflow/pull/9959#discussion_r1369198323

> Is there a reason this is doing some data mutation instead of just validating the data? 🤔 can we make this a util function which doesn't do this? Let me know I am missing something

This is to fix a bug in a hacky way:
If the user provides a single value tuple `("doc1",)` in the dataframe, it will be automatically unpacked into a string here. Ideally, we should fix the bug by not automatically unpack single value tuples. However, we don't know where is this unpacked, so we implement this data manipulation here. Let me know if you have any suggestions here.
@sunishsheth2009